src/library/module.cpp

```diff
@@ -579,7 +580,7 @@ optional<declaration> is_decl_modification(modification const & mod) {
 
 } // end of namespace module
 
-bool is_candidate_olean_file(std::string const & file_name) {
+bool is_candidate_olean_file(std::string const & file_name, unsigned src_hash) {
```

Comment:
We want to check if the olean might be valid for the given source file, so checking for a match of the stored source hash allows rejecting outdated oleans early. It cannot be `trans_hash`, because when `is_candidate_olean_file` is called, we only know the location of the source file but not its imports, so we cannot compute the transitive hash just yet.